.github/workflows/e2e-k8s-main.yaml

```diff
@@ -0,0 +1,145 @@
+
+on:
+  push:
+    branches:
+      - main
+
+  workflow_dispatch:
+    inputs:
+      organization:
+        default: 'podman-desktop'
+        description: 'Organization of the extension-minikube repository'
+        type: string
+        required: true
+      repositoryName:
+        default: 'extension-minikube'
+        description: 'Podman Desktop Minikube extension repository name'
+        type: string
+        required: true
+      branch:
+        default: 'main'
+        description: 'Podman Desktop Minikube extension repo branch'
+        type: string
+        required: true
+      minikube_driver: 
+        description: 'Minikube cluster driver'
+        type: choice
+        options:
+        - docker
+        - podman 
+        required: true
+
+jobs:
+  e2e-tests:
+    name: Run E2E tests
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v4
+        name: Clone Minikube extension repository workflow dispatch
+        if: github.event_name == 'workflow_dispatch'
+        with:
+          repository: ${{ github.event.inputs.organization }}/${{ github.event.inputs.repositoryName }}
+          ref: ${{ github.event.inputs.branch }}
+          path: ${{ github.event.inputs.repositoryName }}
+
+      - uses: actions/checkout@v4
+        with: 
+          path: extension-minikube
+
+      - uses: actions/checkout@v4
+        with:
+          repository: podman-desktop/podman-desktop
+          ref: main
+          path: podman-desktop
+
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+
+      - uses: pnpm/action-setup@v4
+        name: Install pnpm
+        with:
+          run_install: false
+          package_json_file: ./podman-desktop/package.json
+    
+      - name: Set the default minikube cluster driver env. variable
+        env:
+          DEFAULT_MINIKUBE_DRIVER: 'docker'
+        run: |
+          echo "MINIKUBE_DRIVER=${{ github.event.inputs.minikube_driver || env.DEFAULT_MINIKUBE_DRIVER }}" >> $GITHUB_ENV
+
+      - name: Revert unprivileged user namespace restrictions in Ubuntu 24.04
+        run: |
+          # allow unprivileged user namespace
+          sudo sysctl -w kernel.apparmor_restrict_unprivileged_userns=0
+
+      - name: Set cgroup_manager to 'cgroupfs' instead of systemd
+        run: |
+          mkdir -p ~/.config/containers
+          cat <<EOT >> ~/.config/containers/containers.conf
+          [engine]
+          cgroup_manager="cgroupfs"
+          EOT
+          podman info 
+
+      - name: Install Minikube CLI
+        id: minikube
+        uses: medyagh/setup-minikube@latest
+        with: 
+          start: false
+
+      - name: Ensure Minikube CLI is installed
+        run: |
+          minikube version
+            
+      - name: Execute pnpm
+        working-directory: ./podman-desktop
+        run: pnpm install
+
+      - name: Build Podman Desktop for E2E tests
+        working-directory: ./podman-desktop
+        run: pnpm build
```

Comment:
`test:e2e:build`

Reply:
You may consider using testing-prereleases instead of fetching podman-desktop repository and building it locally. https://github.com/podman-desktop/testing-prereleases/tags